torch/nn/modules/module.py

```diff
@@ -20,7 +23,24 @@ def _addindent(s_, numSpaces):
     return s
 
 
-class Module(object):
+class _ModuleMeta(type):
+    """Metaclass for the Module class.
+
+    Saves all `__init__` arguments to `_args` and `_kwargs` module attributes.
+    """
+    def __call__(cls, *args, **kwargs):
+        # Move positional to keyword arguments, but check for name clashes first.
+        # If there are any, call __init__ as-is to get a standard exception.
+        argnames = inspect.getargspec(cls.__init__).args[1:]
+        for name, value in zip(argnames, args):
+            kwargs.setdefault(name, value)
+        args = args[len(argnames):]
+        obj = type.__call__(cls, *args, **kwargs)
+        obj.__dict__.update(_args=args, _kwargs=kwargs)
+        return obj
```

Comment:
Can you make it work only with `torch.*` modules? I'm a bit afraid of supporting it in user-defined modules, since constructors can take in some pretty heavyweight args like pretrained weights, etc.